src/embeddings.rs

```diff
@@ -87,8 +88,20 @@ impl PyEmbeddings {
 
         let results =
             match embeddings.analogy_masked(word1, word2, word3, limit, [mask.0, mask.1, mask.2]) {
-                Some(results) => results,
-                None => return Err(exceptions::KeyError::py_err("Unknown word or n-grams")),
+                Ok(results) => results,
+                Err(lookup) => {
+                    let words = vec![word1, word2, word3];
```

Comment:
Hah, we should have used `[&str; 3]` in `finalfusion-python` as the argument for the words. Saves an auxiliary `Vec` here and the argument is then more synchronous to the mask. Maybe the signature should be:

```
analogy_masked(words: [&str; 3], mask: [bool; 3], limit: usize) -> ...
```

(Or perhaps `impl AsRef<str>` in place of `&str`.)

Reply:
At any rate using an array of size 3 locally should also be more efficient than `vec`, because no extra heap allocations.

Reply:
Doesn't seem like we can use Arrays here: 
```
error[E0277]: the trait bound `[&str; 3]: pyo3::conversion::FromPyObject<'_>` is not satisfied
  --> src/embeddings.rs:33:1
   |
33 | #[pymethods]
   | ^^^^^^^^^^^^ the trait `pyo3::conversion::FromPyObject<'_>` is not implemented for `[&str; 3]`
```

Tuples work tho

Reply:
I meant taking an array in `finalfusion-rust`. If you want to mirror the API in pyo3, you could implement FromPyObject for (a wrapper) of `[&str;3]`. See e.g.:

https://github.com/danieldk/sticker-python/blob/62a10251693ce2ebc053d66109811d37bfc1966f/src/util.rs#L33

(that's to `Vec`, but similar idea)

Reply:
Ah, I got confused. I'll make the PR over at `finalfusion-rust`, let's put this one on ice until that's done?

Reply:
Sounds good, should be a quick & easy change.